mdx-web/src/main/java/com/mx/path/model/mdx/web/controller/AuthenticationController.java

```diff
@@ -215,6 +216,31 @@ public final ResponseEntity<ForgotUsername> forgotUsername(@RequestBody ForgotUs
     return new ResponseEntity<>(response.getResult().wrapped(), createMultiMapForResponse(response.getHeaders()), status);
   }
 
+  @RequestMapping(value = "/unlock_user", method = RequestMethod.POST)
+  public final ResponseEntity<UnlockUser> unlockUser(@RequestBody UnlockUser unlockUser) {
+    //This endpoint always creates a new session when it called even if there is an existing session being passed
+    Session.deleteCurrent();
+    Session.createSession();
+
+    AccessorResponse<UnlockUser> response = gateway().id().unlockUser(unlockUser);
+
+    HttpHeaders headers = new HttpHeaders();
+    headers.add("mx-session-key", Session.current().getId());
+    return new ResponseEntity<>(response.getResult().wrapped(), createMultiMapForResponse(response.getHeaders(), headers), HttpStatus.ACCEPTED);
+  }
+
+  @RequestMapping(value = "/unlock_user", method = RequestMethod.PUT)
```

Comment:
Should there be an id included in this path? Usually PUT requests are used to update a specific resource with the provided id. For `reset_password` and `forgot_username`, the "answer" path includes `/challenges/{challengeId}`. What was the reasoning for not matching the other endpoints that are similar to this?

Reply:
Nevermind. This was a suggestion in the [Gutenberg MR](https://gitlab.mx.com/mx/gutenberg/-/merge_requests/796#note_2862700).